### DISCUSSION TITLE
[general] Does this package lock users into R code only?  if so are there any plans so the package can call other executables

### DISCUSSION THREAD

**Discussion by oTheAnalyst** [ropensci/targets#1270] [General]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

 Are there any plans to add a backend interface so you could run the pipelines on various cluster backends like SLURM, AWS Batch, etc?

**Comment by Aariq** [comment]
You can run targets pipelines on various backends with the `crew` package and `crew.cluster` package including SLURM and AWS.  There are some examples in the [manual](https://books.ropensci.org/targets/crew.html#backends).  

As for running things besides R code, you can create R code wrappers for calling commandline tools with `system2()` or the [`processx` package](https://processx.r-lib.org/), or you can run python code from R with [`reticulate`](https://rstudio.github.io/reticulate/).  I think creating these kinds of wrappers in R is usually your best be for integrating other things besides R with `targets`

**Comment by kendonB** [comment]
I have had some python-based junior colleagues try out snakemake after I had such a good experience with targets. They didn't seem to easily adopt snakemake like my R-based colleagues had been able to pick up targets (and drake before that). If anyone has an opinion, would it be easier / better to try to use targets rather than using snakemake for someone who mostly uses python?

**Reply by ginolhac** [reply]
Snakemake is useful and for bioinformatics a must have.
But building dependencies from the bottom (rule all) and wildcards are making writing of new pipelines much more difficult than in targets. 
Also it is lacking the debugging facilities of targets.
For users, good templates are very nice to have and use. I guess this is the core aspect of snakemake.
Targets is more about writing your analysis with clear dependencies than applying recipes.
